# Graded assignment 1

Mario van Kessel


# Packages

We need the package `tidyverse`

In [3]:
library(tidyverse) 

-- Attaching packages --------------------------------------- tidyverse 1.2.1 --
v ggplot2 2.2.1     v purrr   0.2.4
v tibble  1.4.1     v dplyr   0.7.4
v tidyr   0.7.2     v stringr 1.2.0
v readr   1.1.1     v forcats 0.2.0
-- Conflicts ------------------------------------------ tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()


# Assignment 1

First, we generate some random data:

In [4]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3        V4        V5        V6        V7       
1 0.2416850 0.8228349 0.5547694 0.5228904 0.6916123 0.8646104 0.8370648
2 0.3898285 0.2290497 0.5030750 0.4368944 0.5726964 0.5509405 0.4185862
  V8         V9        V10       V11       V12       V13       V14       
1 0.94916551 0.1665715 0.7242841 0.1398440 0.9370645 0.3040916 0.06954118
2 0.08272662 0.7025568 0.2239659 0.5132317 0.2603306 0.4434378 0.78905798
  V15       V16       V17        V18       V19       V20      
1 0.5772996 0.9283773 0.21933320 0.3679788 0.3910855 0.5871950
2 0.6255680 0.7985343 0.03672122 0.9870983 0.9907351 0.4967537

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [5]:
df2 <- df1 %>% gather(Key, Value, V1:V20)
  
head(df2)

Key Value    
1 V1  0.2416850
2 V1  0.3898285
3 V2  0.8228349
4 V2  0.2290497
5 V3  0.5547694
6 V3  0.5030750

## Assignment 2

We will generate some data:

In [6]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [7]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [8]:
df5 <- merge(df3, df4, all.x=TRUE, all.y=TRUE )
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook